exporter/importables.go

```diff
@@ -856,4 +856,57 @@ var resourcesMap map[string]importable = map[string]importable{
 			return nil
 		},
 	},
+	"databricks_repo": {
+		Service: "repos",
+		Name: func(d *schema.ResourceData) string {
+			name := d.Get("path").(string)
+			if name == "" {
+				return d.Id()
+			} else {
+				name = strings.TrimPrefix(name, "/")
+			}
+			re := regexp.MustCompile(`[^0-9A-Za-z_]`)
+			return re.ReplaceAllString(name, "_")
+		},
+		List: func(ic *importContext) error {
+			repoList, err := workspace.NewReposAPI(ic.Context, ic.Client).ListAll()
+			if err != nil {
+				return err
+			}
+			for offset, repo := range repoList {
+				if repo.Url != "" {
+					ic.Emit(&resource{
+						Resource: "databricks_repo",
+						ID:       fmt.Sprintf("%d", repo.Id),
+					})
+				}
+				log.Printf("[INFO] Scanned %d of %d repos", offset+1, len(repoList))
+			}
+			return nil
+		},
+		Import: func(ic *importContext, r *resource) error {
+			if ic.meAdmin {
+				ic.Emit(&resource{
+					Resource: "databricks_permissions",
+					ID:       fmt.Sprintf("/repos/%s", r.ID),
+					Name:     "repo_" + ic.Importables["databricks_repo"].Name(r.Data),
+				})
+			}
+			return nil
+		},
+		Body: func(ic *importContext, body *hclwrite.Body, r *resource) error {
```

Comment:
Why do we need custom body exporter?

Reply:
Because by default it omits fields that are marked as optional, i.e. exporting only URL

Reply:
@alexott probably a separate exporter bug.

Reply:
I think it’s a combination of optional and computed properties. From exporter side we don’t know everything, we need to think on how to handle that correctly